### PR TITLE
Update ruby to 3.1.2

### PR DESCRIPTION
**Pull request by stevieing**
Closes #

#### Changes proposed in this pull request

- 

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: db/migrate/20230522131908_add_library_tube_barcode_to_pacbio_run.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def down
       t.remove :pac_bio_library_tube_barcode
     end
     # Here we add a placeholder value in case any plate barcodes were added as null since the migration
-    PacBioRun.where(plate_barcode: nil).each do |run|
```
Comment (BenTopping) [inline db/migrate/20230522131908_add_library_tube_barcode_to_pacbio_run.rb:17]: I would add 'db/migrate/*' to the rubocop ignore list as we don't need to bother updating old migrations.

**Review by BenTopping — COMMENTED** [review]
There is a docker image for unified warehouse, although I don't know where its used. The Dockerfile will need updating to reflect the correct ruby version.

File: lib/resource_tools/json.rb [hunk 2]
```diff
@@ -156,6 +157,7 @@ def initialize(*args, &block)
       convert_booleans
       delete_if { |k, _| ignoreable.include?(k) }
     end
+    # rubocop:enable Naming/BlockForwarding
```
[the remark this answers is not in the document]
Comment (stevieing) [reply lib/resource_tools/json.rb:160]: The rubocop change was causing test failures. Didn't really want to get into fixing it.

**Comment by stevieing** [comment]
> There is a docker image for unified warehouse, although I don't know where its used. The Dockerfile will need updating to reflect the correct ruby version.

This is concerning. It means we don't have enough coverage as this is not immediately obvious that it needs changing but would cause a breaking change.

File: db/migrate/20230522131908_add_library_tube_barcode_to_pacbio_run.rb (hunk 1 above)
Comment (stevieing) [reply db/migrate/20230522131908_add_library_tube_barcode_to_pacbio_run.rb:17]: Agreed. I will leave the change but add it to the gitignore.

File: lib/resource_tools/json.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (yoldas) [reply lib/resource_tools/json.rb:160]: I looked at the Rubocop suggestions, which are not right. custom_values contains name => block entries. The suggested fix breaks that and mixes it with the other block parameter and ignores the block value in the iteration. I think, ignoring the rules is the right thing to do here as it has been done.

**Review by yoldas — APPROVED** [review]

**Review by BenTopping — APPROVED** [review]

**Pull request merged**